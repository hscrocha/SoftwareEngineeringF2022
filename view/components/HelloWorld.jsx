import React from "react";
import { Text, View } from "react-native";

export default function HelloWorld(props){
    let message = "Hello World";
    if(props.children) message = props.children;
    else if(props.text) message = props.text;
    else if(props.route.params.text) message = props.route.params.text;

    return (
        <View style={{flex: 1,justifyContent:'center',alignItems:'center'}}>
            <Text style={{fontSize:20}}> {message} </Text>
        </View>
    );
}