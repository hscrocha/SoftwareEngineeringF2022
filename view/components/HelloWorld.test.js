import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HelloWorld from "./HelloWorld";

test('Hello World Comp. Snapshot Test',function(){
    const rnElement = renderer.create(<HelloWorld />).toJSON();
    expect(rnElement).toMatchSnapshot();
});