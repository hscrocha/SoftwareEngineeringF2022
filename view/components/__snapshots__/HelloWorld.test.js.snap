// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World Comp. Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
     
    Hello World
     
  </Text>
</View>
`;
